docs/packages/context-menu/Context-menu-manager.md

```diff
@@ -0,0 +1,345 @@
+
+### Context menu UI behavior
+For better understanding the structure and the manager, lets see some important UI notes:
+
+* The context menu displays a list of items, by default each item provides an action when clicking on it.
+
+* Items can provide action also on hover (mouseEnter).
+
+* If an item contains children groups, hovering on it will open a sub menu, and clicking on it won't trigger action.
+
+* Groups themselves are not visually displayed, they just displays their items and separate between them.
+
+* A group can display a title, if a title is specified.
+
+* Items of a group are displayed together, separated from other group items with a line separator.
+
+* Nested groups are displayed as a sub-menu aside the main context-menu.
+
+### Context Menu config structure
+The context menu config structure is built as a tree of groups & items.
+The root is based on groups, each group contains one or more item (inside its `items[]` property).
+Items can also have children, inside their `groups[]` property. those children are *nested groups*.
+
+### Structure tree
+
+```TS
+interface Group {
+	id: string; 				// Must be unique.
+	title: string;				// If exists, displayed as a group title without action to trigger.
+	disabled: Function | boolean;		// If true, the entire group is Displayed but disabled.
+	parentItemId: string;			// Id of the parent (a valid item id). if used - creates a nested menu.
+	items: Item[]; 				// List of items.
+};
+
+interface Item {
+	id: string;				// Must be unique.
+	label: string;				// Describes the action.
+	icon: Function | string;		// Can be a callback for checkbox effect.
+	shortCut: string;			// ShortCut symbols. 
+	badge: Badge;				// Creates a visual badge for special indications.
+	disabled: Function | boolean;		// If true, displayed but disabled.
+	action: Function;			// A callback to be triggered on-click.
+	onMouseEnter: Function;			// A callback to be triggered on-mouse-enter.
+	onMouseLeave: Function;			// A callback to be triggered on-mouse-leave.
+	groups: Group[];			// List of groups (for nested menus).
+};
+
+interface Badge {
+	backgroundColor: string;		// Background color of the badge.
+	color: string;				// Text color of the badge.
+	text: string;				// Text of badge.
+}
+
+type Structure = Group[];
+```
+
+
+&nbsp;  
+## API methods
+Each method has parameters explanation and attention notes if needed.
+
+&nbsp;
+### Create a group
+`createGroup( newGroup )` Method creates a group and add it to the config tree. 
+
+&nbsp;  
+Example:
+```JS
+createGroup(
+ 	{
+ 		id: '',				// Required.
+ 		title: '', 			// Optional, if not specified will be '' as default.
+ 		disabled: false,	// Optional, if not specified will be false as default.
+ 		parentItemId: '',	// Optional, if not specified will be '' as default.
```

Comment:
if it's optional, define that in the Type above

Reply:
It is optional to specify those properties when using the API method. 
but if not specified, the manager will fill them with default values.